nix/default.nix

```diff
@@ -3,34 +3,31 @@
 , config ? { }
 , overlays ? [ ]
 , sourcesOverride ? { }
+, sources
+, isInFlake
+, haskellNix
 , checkMaterialization ? false
 , enableHaskellProfiling ? false
 }:
 let
-  sources = import ./sources.nix { inherit pkgs; }
-    // sourcesOverride;
-  haskellNix = import sources."haskell.nix" {
-    sourcesOverride = {
-      hackage = sources."hackage.nix";
-      stackage = sources."stackage.nix";
-    };
-  };
-
-  extraOverlays =
-    # Haskell.nix (https://github.com/input-output-hk/haskell.nix)
-    haskellNix.overlays
-    # our own overlays:
-    ++ [
+  ownOverlays =
+    [
       # Modifications to derivations from nixpkgs
       (import ./overlays/nixpkgs-overrides.nix)
       # fix r-modules
       (import ./overlays/r.nix)
     ];
 
+  # haskell-nix has to be used differently in flakes/no-flakes scenarios:
+  # - When imported from flakes, 'haskellNix.overlay' needs to be passed here.
+  # - When imported from default.nix without flakes, default to haskellNix.overlays
```

Comment:
Can you also open an issue upstream about this? it's dumb

Reply:
See https://github.com/input-output-hk/haskell.nix/issues/1130